lib/auth0/mixins/httpproxy.rb

```diff
@@ -7,11 +7,12 @@ module HTTPProxy
 
       # proxying requests from instance methods to HTTP class methods
       %i(get post post_file put patch delete).each do |method|
-        define_method(method) do |path, body = {}|
+        define_method(method) do |path, body = {}, extra_headers = {}|
           safe_path = URI.escape(path)
           body = body.delete_if { |_, v| v.nil? }
           result = if [:get, :delete].include?(method)
-                     call(method, url(safe_path), timeout, add_headers(params: body))
+                     add_headers(extra_headers.nil? ? {params: body} : extra_headers )
```

Comment:
Is this usage of `add_headers` compatible with the existing one? Does `add_headers` return a value or does it mutate something?

Reply:
> Is this usage of `add_headers` compatible with the existing one? 

Yes. That's there to ... add headers but the current implementation uses it to add URL params by setting `:params` to be the `body` that's passed in for `GET` requests. It's a bit confusing in it's current state but I think this is a good compromise between necessary functionality and legacy support. 

> Does `add_headers` return a value or does it mutate something?

Both. It mutates the module's `@headers` property and returns it. I'm using it just to mutate, then pass the property to that `call()` method.